src/Common/src/Interop/Windows/kernel32/Interop.GetConsoleTitle.cs

```diff
@@ -5,68 +5,66 @@
 using System.Diagnostics;
 using System.Text;
 using System.Runtime.InteropServices;
+using System;
 
 internal partial class Interop
 {
     internal partial class Kernel32
     {
-        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "GetConsoleTitleW")]
-        private static extern int GetConsoleTitle([Out]StringBuilder title, int nSize);
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, SetLastError = true, ExactSpelling = true)]
+        private static extern uint GetConsoleTitleW(ref char title, uint nSize);
 
-        private const int InitialBufferSizeInChars = 256;
+        private static bool s_isWindows7 = GetIsWindows7();
 
-        // Although msdn states that the max allowed limit is 65K,
-        // desktop limits this to 24500 as buffer sizes greater than it
-        // throw.
-        private const int MaxAllowedBufferSizeInChars = 24500;
-
-        // GetConsoleTitle sometimes interprets the second parameter (nSize) as number of bytes and sometimes as the number of chars.
-        // Instead of doing complicated and dangerous logic to determine if this may or may not occur,
-        // we simply assume the worst and reserve a bigger buffer. This way we may use a bit more memory,
-        // but we will always be safe.
-        private static int CharCountToByteCount(int numChars)
+        private static bool GetIsWindows7()
         {
-            return numChars * 2;
+            // Version lies from the OS kick in starting with Windows 8 (6.2)
```

Comment:
The version lies are pretty fragile. I think that the original logic that just reserved bigger buffer was much more robust workaround for the OS bug.

Reply:
I'm going to adjust this to do this safely.